test/old_tests/UnitTests/delegate_weak_strong.cpp

```diff
@@ -81,6 +113,51 @@ namespace
         weak({}, {});
         REQUIRE(weak_count == 2);
     }
+
+    template <typename Delegate, typename Sender, typename Args>
+    void test_delegate_std()
+    {
+        auto object = std::make_shared<ObjectStd<Sender, Args>>();
+
+        Delegate strong{ object->shared_from_this(), &ObjectStd<Sender, Args>::StrongHandler };
+        Delegate weak{ object->weak_from_this(), &ObjectStd<Sender, Args>::WeakHandler };
+
+        destroyed = false;
+        strong_count = 0;
+        weak_count = 0;
+
+        // Both weak and strong handlers
+        strong({}, {});
+        weak({}, {});
+        REQUIRE(strong_count == 1);
+        REQUIRE(weak_count == 1);
+
+        // Local 'object' strong reference is released
+        object = nullptr;
+
+        // Still both since strong handler keeps object alive
+        strong({}, {});
+        weak({}, {});
+        REQUIRE(strong_count == 2);
+        REQUIRE(weak_count == 2);
+
+        // ~Object is called since the strong delegate is destroyed
+        REQUIRE(!destroyed);
+        strong = nullptr;
+        REQUIRE(destroyed);
```

Comment:
>destroyed [](http://example.com/codeflow?start=16&length=9)

Preexisting, but I'm confused by how the struct implementation is aliasing local variables in this method without any sort of obvious capturing.

Reply:
They're global variables, declared at the top of this file.

Reply:
Whoops, overlooked that detail.